app/controllers/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+class UsersController < ApplicationController
+
+  get '/users/:slug' do
+      @user = User.find_by_slug(params[:slug])
+      erb :'users/show'
+  end
+  
+  get '/signup' do
+    if !is_logged_in?
+      erb :'/users/create_user', locals: {message: "Please sign up before you sign in"}
+    else
+     redirect to '/tweets'
+    end
+  end
+
+  post '/signup' do
+    @user = User.new(params)
+    if @user.username != "" && @user.email != "" && @user.password != "" && @user.password != nil
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.